### ISSUE TITLE
Z-Image training preview images ignore `guidance` parameter from config

### ISSUE THREAD

**Issue by azrahello** [filipstrand/mflux#347]
# Bug: Z-Image training preview images ignore `guidance` parameter from config

## Description

When training a LoRA on `z-image` (non-turbo) with `guidance` set in the training config JSON, the preview images generated during training always use `guidance=0.0` instead of the configured value. This results in preview images that do not accurately represent the actual training progress, since Z-Image standard relies on Classifier-Free Guidance for quality output.

## Root Cause

In `z_image_training_adapter.py`, the `generate_preview_image` method calls `self._z.generate_image()` without passing the `guidance` parameter:

```python
def generate_preview_image(self, *, seed, prompt, width, height, steps, image_paths=None):
    with self._assistant_disabled():
        return self._z.generate_image(
            seed=seed, prompt=prompt, num_inference_steps=steps, height=height, width=width
        )
```

Since `guidance` is not passed, `generate_image` receives `guidance=None`, which is then converted to `0.0` in `Config.__init__`:

```python
self._guidance = 0.0 if guidance is None else float(guidance)
```

With `guidance=0.0`, `_encode_prompts` skips negative encoding entirely (`if guidance <= 1.0: return text_encodings, None`), so no CFG is applied.

## Why it only affects Z-Image (non-turbo)

Z-Image Turbo has `supports_guidance=False` in its `ModelConfig`, so `generate_image` forces `guidance=0.0` regardless — the bug is invisible. Z-Image standard has `supports_guidance=True` and requires CFG (`guidance > 1.0`) for proper generation quality.

## Steps to Reproduce

1. Create a training config with `"model": "z-image"` and `"guidance": 5.0`
2. Start training with `mflux-train`
3. Observe that preview images are generated with `guidance=0.0` (visible in the output metadata or by comparing quality with a manually generated image using `guidance=5.0`)

## Suggested Fix

Cache the `guidance` value from the training spec and pass it to `generate_image`:

```python
class ZImageTrainingAdapter(TrainingAdapter):
    def __init__(self, *, model_config: ModelConfig, quantize: int | None):
        self._model_config = model_config
        self._z = ZImage(quantize=quantize, model_config=model_config)
        self._guidance: float | None = None

    def create_config(self, training_spec: TrainingSpec, *, width: int, height: int) -> Config:
        if self._guidance is None and training_spec.guidance is not None:
            self._guidance = training_spec.guidance
        return Config(
            model_config=self._model_config,
            num_inference_steps=training_spec.steps,
            width=width,
            height=height,
            guidance=training_spec.guidance,
        )

    def generate_preview_image(self, *, seed, prompt, width, height, steps, image_paths=None):
        with self._assistant_disabled():
            return self._z.generate_image(
                seed=seed, prompt=prompt, num_inference_steps=steps, height=height, width=width,
                guidance=self._guidance,
            )
```

Alternatively, the `generate_preview_image` protocol in `TrainingAdapter` could be extended to accept a `guidance` parameter, propagated from the trainer.

## Environment

- mflux version: 0.16.2
- Model: `z-image` (Tongyi-MAI/Z-Image)
- Hardware: Apple M2 Ultra, 64GB

**Comment by filipstrand** [comment]
@azrahello Thanks for finding this! This is something I forgot to test myself so not totally surprised it didn't work :) Will push an update for this soon

**Comment by azrahello** [comment]
Another thing that seems odd to me is the resource usage during the first generation. Shouldn't it use the same amount of RAM as a normal generation? During the training phase, on the other hand, it starts swapping right away. PS: in the test phase, this config should work:
json{
  "model": "z-image-turbo",
  "data": "images/",
  "seed": 4,
  "steps": 9,
  "guidance": 0.0,
  "quantize": 8,
  "training_loop": { "num_epochs": 1, "batch_size": 1, "timestep_low": 4, "timestep_high": 9 },
  "optimizer": { "name": "AdamW", "learning_rate": 1e-4 },
  "checkpoint": { "output_path": "training", "save_frequency": 30 },
  "monitoring": {
    "plot_frequency": 1,
    "generate_image_frequency": 30
  },
  "lora_layers": {
    "targets": [
      { "module_path": "layers.{block}.attention.to_q", "blocks": { "start": 15, "end": 30 }, "rank": 8 },
      { "module_path": "layers.{block}.attention.to_k", "blocks": { "start": 15, "end": 30 }, "rank": 8 },
      { "module_path": "layers.{block}.attention.to_v", "blocks": { "start": 15, "end": 30 }, "rank": 8 }
    ]
  }
}
But for me, this config can't even complete the first step. Adding max_resolution makes it work instead
